File: qa/page/LoginPage.ts
import { Page } from '@playwright/test';

export class LoginPage {
  private page: Page;

  // Selectors
  private nameInput = 'input[placeholder="Name"]';
  private emailInput = 'input[data-qa="signup-email"]';
  private signupButton = 'button:has-text("Signup")';

  private loginEmailInput = '//input[@data-qa="login-email"]';
  private loginPasswordInput = '//input[@data-qa="login-password"]';
  private loginButton = '//button[@data-qa="login-button"]';


  constructor(page: Page) {
    this.page = page;
  }

  // Navigate to the Signup Page
  async navigate() {
    await this.page.goto('https://automationexercise.com/login');
  }

  // Perform Signup
  async signup(name: string, email: string) {
    await this.page.fill(this.nameInput, name);
    await this.page.fill(this.emailInput, email);
    await this.page.click(this.signupButton);
    await this.page.waitForTimeout(2000);
  }

  // Verify successful signup navigation
  async verifySignupPage() {
    await this.page.waitForURL(/signup/, { timeout: 5000 });
  }

  // Perform Login
  async login(email: string, password: string) {
    await this.page.fill(this.loginEmailInput, email); 
    await this.page.fill(this.loginPasswordInput, password); 
    await this.page.click(this.loginButton);
  }
}

File: qa/java-playwright/src/test/java/ui/pages/LoginPage.java
package ui.pages;

import com.microsoft.playwright.Page;

import static com.microsoft.playwright.assertions.PlaywrightAssertions.assertThat;

public class LoginPage {
    private final Page page;

    public LoginPage(Page page) {
        this.page = page;
    }

    public void open() {
        page.navigate("https://automationexercise.com/login");
        assertThat(page.locator("text=Login to your account")).isVisible();
    }

    public void login(String email, String password) {
        page.locator("input[data-qa='login-email']").fill(email);
        page.locator("input[data-qa='login-password']").fill(password);
        page.locator("button[data-qa='login-button']").click();
    }
}
